### PR TITLE
Update boilerplate code to the latest version

### PR DESCRIPTION
**Pull request by pienkowb**

### CODE REVIEW

**Review by ybce — APPROVED** [review]

File: src/handle_init_contract.c [hunk 1]
```diff
@@ -35,17 +23,24 @@ void handle_init_contract(void *parameters) {
...
+    if ((size_t) context->selectorIndex != index) {
+        PRINTF("Error: overflow detected on selector index!\n");
+        msg->result = ETH_PLUGIN_RESULT_ERROR;
+        return;
+    }
 
     // Set `next_param` to be the first field we expect to parse.
     switch (context->selectorIndex) {
         case FIGMENT_DEPOSIT:
+            context->next_param = PUBKEYS_ARRAY;
```
Comment (pienkowb) [inline src/handle_init_contract.c:43]: Previously, this was set implicitly (by zeroing the `context`).

File: src/handle_provide_token.c [hunk 2]
```diff
@@ -0,0 +1,5 @@
+#include "plugin.h"
+
+void handle_provide_token(ethPluginProvideInfo_t *msg) {
+    msg->result = ETH_PLUGIN_RESULT_OK;
+}
```
Comment (pienkowb) [inline src/handle_provide_token.c:5]: Defining this method is now required (but we don't need use it in our plugin).

File: src/handle_init_contract.c [hunk 3]
```diff
@@ -35,17 +23,24 @@ void handle_init_contract(void *parameters) {
     // Initialize the context (to 0).
     memset(context, 0, sizeof(*context));
 
-    uint32_t selector = U4BE(msg->selector, 0);
-    if (find_selector(selector, FIGMENT_SELECTORS, NUM_SELECTORS, &context->selectorIndex)) {
+    size_t index;
+    if (!find_selector(U4BE(msg->selector, 0), SELECTORS, SELECTOR_COUNT, &index)) {
+        PRINTF("Error: selector not found!\n");
```
Comment (mkornatz) [inline src/handle_init_contract.c:28]: Are printf's ok to have in here, or do we need to write the message somewhere else?

**Review by mkornatz — APPROVED** [review]
Looks OK to me. Just have one question about printf statements

File: src/handle_init_contract.c (hunk 3 above)
Comment (pienkowb) [reply src/handle_init_contract.c:28]: This part comes from the boilerplate plugin. I think it's the recommended way of reporting such errors.

**Pull request merged**